osquery/tables/system/windows/chocolatey_packages.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#include <boost/filesystem.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/process.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+Status genPackage(fs::path nuspec, Row& r) {
+  std::string content;
+  if (!readFile(nuspec, content).ok()) {
+    return Status(1, "Failed to read nuspec:" + nuspec.string());
+  }
+
+  std::stringstream ss;
+  ss << content;
+  pt::ptree propTree;
+  read_xml(ss, propTree);
```

Comment:
This may throw, do you want to handle it here, or allow the registry to catch it?

Reply:
If you handle it, and return a 1 status you will allow the table to emit partial results.